tsweb/tsweb.go

```diff
@@ -204,6 +203,13 @@ type ErrorHandlerFunc func(http.ResponseWriter, *http.Request, HTTPError)
 // calls f.
 type ReturnHandlerFunc func(http.ResponseWriter, *http.Request) error
 
+// Middleware is a function which receives an http.Handler and returns a
+// modified http.Handler. Typically, the returned handler is a closure which
+// does something with the http.ResponseWriter and *http.Request that are passed
+// to it, and then calls the handler passed as parameter to the original
+// Middleware.
+type Middleware func(h http.Handler) http.Handler
```

Comment:
I think we should distinguish between the documentation and explanation parts of the comment a bit more clearly. This is a kind of interface, and a lot of what's said here is a prescription for the implementation of it, rather than something the middleware plumbing enforces by itself.

Perhaps:

```go
// A Middleware is a function that wraps an http.Handler to extend or modify
// its behaviour.
//
// The implementation of the wrapper is responsible for delegating its input
// request to the underlying handler, if appropriate.
```

(note that nothing requires a Middleware to actually call the underlying handler; you could imagine a "filtering" middleware that only does so conditionally)

Reply:
Good point 🙏🏼 updated